dvc/cache/base.py

```diff
@@ -520,7 +520,9 @@ def checkout(
             self.safe_remove(path_info, force=force)
             failed = path_info
 
-        elif not relink and not self.changed(path_info, hash_info):
+        elif not relink and not self.changed(
+            filter_info or path_info, hash_info
+        ):
```

Comment:
hash_info is a hash_info of the full dataset, right? But filter_info might be about a subset. Please check `changed()` method, I don't think it is ready for this.

Reply:
You are right. In 323d20d, I tried to address by using filter_info in changed if the checks related to paths, and using the original path_info otherwise (like comparing the hashs)